lsp-mode.el

```diff
@@ -9934,6 +9957,43 @@ string."
     (setf window-scroll-functions
           (delete #'lsp--update-inlay-hints-scroll-function window-scroll-functions)))))
 
+
+;; Inline Completions
+(defvar-local lsp-inline-completion--idle-timer nil
+  "The idle timer used by lsp-inline-completion-mode")
+
+(define-minor-mode lsp-inline-completion-mode
+  "Mode automatically displaying inline completions."
+  :lighter nil
+  (cond
+   ((and lsp-inline-completion-mode lsp--buffer-workspaces)
```

Comment:
Can we use `lsp-on-idle-hook` instead? The `lsp-inlay-hints-mode` is using that and it has been working well so far. It also should fire less frequently compared to `on-change-hook` as when typing, we're likely don't want to request inline completion anyway.

Reply:
Hmm, I don't think that's a good idea.

I've tested here, and triggering it via `lsp-idle-hook` means we get suggestions after moving the cursor around -- I don't think a user would expect that (I, for one, don't).

Regarding the change hook being more aggressive than idle: Since lsp's idle timer is set both on `post-command-hook` and on `after-change-hook`, I don't see how that could be.

In my implementation, what actually happens in `after-change` is a `cancel-timer` and `run-with-timer` (i.e. timer reschedule). We don't ping the server after each change -- we ping the server after `lsp-inline-completion-idle-delay` seconds have elapsed without any change in the buffer.

Reply:
Ah okay. The Visual Studio will trigger inline completion on mouse move as well while the VS Code seems to only trigger it on buffer modification. Whichever as default is fine for me :)

Reply:
> Visual Studio will trigger inline completion on mouse move as well

It does!?! THE HORRORS!!! :P 

> VS Code seems to only trigger it on buffer modification

We could add a custom variable to control which hook is used, but I would wait for a feature request to avoid implementing something that'd never be used ;) -- If you think you would prefer to have it activated on idle, I can consider this a first feature request and do it in this PR ;)